barretenberg/cpp/src/barretenberg/commitment_schemes/commitment_key.hpp

```diff
@@ -103,6 +106,83 @@ template <class Curve> class CommitmentKey {
+    /**
+     * @brief Batch commitment to multiple polynomials
+     * @details Uses batch_multi_scalar_mul for more efficient processing when committing to multiple polynomials
+     *
+     * @param polynomials vector of polynomial spans to commit to
+     * @return std::vector<Commitment> vector of commitments, one for each polynomial
+     */
+    std::vector<Commitment> batch_commit(RefSpan<Polynomial<Fr>> polynomials,
+                                         size_t max_batch_size = std::numeric_limits<size_t>::max()) const
+    {
+        BB_BENCH_NAME("CommitmentKey::batch_commit");
+        std::span<const G1> point_table = srs->get_monomial_points();
+
+        // We can only commit max_batch_size at a time
+        // This is to prevent excessive memory usage in the pippenger algorithm
+
+        // First batch, create the commitments vector
+        std::vector<Commitment> commitments;
+
+        for (size_t i = 0; i < polynomials.size();) {
+            // Note: have to be careful how we compute this to not overlow e.g. max_batch_size + 1 would
+            size_t batch_size = std::min(max_batch_size, polynomials.size() - i);
+            size_t batch_end = i + batch_size;
+
+            // Prepare spans for batch MSM
+            std::vector<std::span<const G1>> points_spans;
+            // Note, we need to const_cast unfortunately as pippenger takes non-const spans
+            // as it converts back and forth from montgomery form
+            std::vector<std::span<Fr>> scalar_spans;
+
+            for (auto& polynomial : polynomials.subspan(i, batch_end - i)) {
+                std::span<const G1> point_table = srs->get_monomial_points().subspan(polynomial.start_index());
+                size_t consumed_srs = polynomial.start_index() + polynomial.size();
+                if (consumed_srs > srs->get_monomial_size()) {
+                    throw_or_abort(format("Attempting to commit to a polynomial that needs ",
+                                          consumed_srs,
+                                          " points with an SRS of size ",
+                                          srs->get_monomial_size()));
+                }
+                scalar_spans.emplace_back(polynomial.coeffs());
+                points_spans.emplace_back(point_table);
+            }
+
+            // Perform batch MSM
+            auto results = scalar_multiplication::MSM<Curve>::batch_multi_scalar_mul(points_spans, scalar_spans, false);
+            for (const auto& result : results) {
+                commitments.emplace_back(result);
+            }
+            i += batch_size;
+        }
+        return commitments;
+    };
+
+    // helper builder struct for constructing a batch to commit at once
+    struct CommitBatch {
+        CommitmentKey* key;
+        RefVector<Polynomial<Fr>> wires;
+        std::vector<std::string> labels;
+        void send_to_verifier(auto transcript, size_t max_batch_size = std::numeric_limits<size_t>::max())
```

Comment:
can we name this `commit_and_send_to_verifier()` or something? otherwise when used in the wild its not clear where the committing is actually happeneing